lib/FeedItem.php

```diff
@@ -428,6 +428,37 @@ public function setUid($uid) {
 		return $this;
 	}
 
+	/**
+	 * Get duration (for items where the link points to media)
+	 *
+	 * Use {@see FeedItem::setDuration()} to set the duration.
+	 *
+	 * @return int Duration in seconds.
+	 */
+	public function getDuration() {
+		return $this->duration;
+	}
+
+	/**
+	 * Set duration (for items where the link points to media)
+	 *
+	 * Use {@see FeedItem::getDuration()} to get the duration.
+	 *
+	 * @param int $duration Duration of the media in seconds
+	 * @return self
+	 */
+	public function setDuration($duration) {
```

Comment:
You can ensure non-ints are not passed using typehint.

```suggestion
	public function setDuration(int $duration): self {
```

Maybe also allow clearing the value:

```suggestion
	public function setDuration(?int $duration): self {
```